tests/mock_tests/mock_sai_api.h

```diff
@@ -178,6 +178,69 @@ The macro DEFINE_SAI_API_MOCK will perform the steps to mock the SAI API for the
         delete mock_sai_##sai_api_name##_api;                                                                                \
     }
 
+/* Helper macros to iterate over multiple sai_object_type inputs */
+#define FOR_EACH_1(action, api_name, x) action(api_name, x)
+#define FOR_EACH_2(action, api_name, x, ...) action(api_name, x) FOR_EACH_1(action, api_name, __VA_ARGS__)
+
+#define GET_FOR_EACH_MACRO(_1,_2,NAME,...) NAME
+#define FOR_EACH(action, api_name, ...) \
+    GET_FOR_EACH_MACRO(__VA_ARGS__, FOR_EACH_2, FOR_EACH_1)(action, api_name, __VA_ARGS__)
+
+#define DEFINE_ON_CALL_DEFAULTS(sai_api_name, sai_object_type) \
+    ON_CALL(*this, create_##sai_object_type).WillByDefault([this](GENERIC_CREATE_PARAMS(sai_object_type)) { \
+        return old_sai_##sai_api_name##_api->create_##sai_object_type(GENERIC_CREATE_ARGS(sai_object_type)); \
+    }); \
+    ON_CALL(*this, remove_##sai_object_type).WillByDefault([this](GENERIC_REMOVE_PARAMS(sai_object_type)) { \
+        return old_sai_##sai_api_name##_api->remove_##sai_object_type(GENERIC_REMOVE_ARGS(sai_object_type)); \
+    }); \
+    ON_CALL(*this, set_##sai_object_type##_attribute).WillByDefault([this](sai_object_id_t oid, const sai_attribute_t *attr) { \
+        return old_sai_##sai_api_name##_api->set_##sai_object_type##_attribute(oid, attr); \
+    });
+#define DEFINE_MOCK_METHODS(sai_api_name, sai_object_type) \
+    MOCK_METHOD4(create_##sai_object_type, sai_status_t(GENERIC_CREATE_PARAMS(sai_object_type))); \
+    MOCK_METHOD1(remove_##sai_object_type, sai_status_t(GENERIC_REMOVE_PARAMS(sai_object_type))); \
+    MOCK_METHOD2(set_##sai_object_type##_attribute, sai_status_t(sai_object_id_t, const sai_attribute_t *));
+#define DEFINE_WRAPPER_FUNCTIONS(sai_api_name, sai_object_type) \
+    inline sai_status_t mock_create_##sai_object_type(GENERIC_CREATE_PARAMS(sai_object_type)) { \
+        return mock_sai_##sai_api_name##_api->create_##sai_object_type(GENERIC_CREATE_ARGS(sai_object_type)); \
+    } \
+    inline sai_status_t mock_remove_##sai_object_type(GENERIC_REMOVE_PARAMS(sai_object_type)) { \
+        return mock_sai_##sai_api_name##_api->remove_##sai_object_type(GENERIC_REMOVE_ARGS(sai_object_type)); \
+    } \
+    inline sai_status_t mock_set_##sai_object_type##_attribute(sai_object_id_t oid, const sai_attribute_t *attr) { \
+        return mock_sai_##sai_api_name##_api->set_##sai_object_type##_attribute(oid, attr); \
+    }
+#define APPLY_MOCK_FUNCTIONS(sai_api_name, sai_object_type) \
+    sai_##sai_api_name##_api->create_##sai_object_type = mock_create_##sai_object_type; \
+    sai_##sai_api_name##_api->remove_##sai_object_type = mock_remove_##sai_object_type; \
+    sai_##sai_api_name##_api->set_##sai_object_type##_attribute = mock_set_##sai_object_type##_attribute;
+
+
+
+#define DEFINE_SAI_GENERIC_APIS_MOCK(sai_api_name, ...) \
```

Comment:
What's the difference between this macro and `DEFINE_SAI_GENERIC_API_MOCK`, is it just allow mocking multiple APIs with one macro call?

Reply:
yes, we have `ha_set` and `ha_scope` in the same `sai_dash_ha_api`, that's why I need this macro.

Reply:
Ah I see. This should also still work for mocking just a single object type right? Can you add a comment above the old `DEFINE_SAI_GENERIC_API_MOCK` macro to deprecate it and use your new macro instead?

Reply:
Addressed.